### PR TITLE
feat(plasma-web): Navigate by disabled option

### PR DESCRIPTION
**Pull request by Yakutoc**
В компоненте Select -> Drop Down необходимо уметь навигировать по выключенным элементам.        

:baby_chick: Download canary assets:

[colors--canary.269.3780484831.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.269.3780484831.xml)  
[color_metro_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.269.3780484831.swift)  
[color_metro_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.269.3780484831.kt)  
[color_metro_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.269.3780484831.ts)  
[color_sbermarket_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.269.3780484831.swift)  
[color_sbermarket_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.269.3780484831.kt)  
[color_sbermarket_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.269.3780484831.ts)  
[color_sberprime_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.269.3780484831.swift)  
[color_sberprime_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.269.3780484831.kt)  
[color_sberprime_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.269.3780484831.ts)  
[color_selgros_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.269.3780484831.swift)  
[color_selgros_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.269.3780484831.kt)  
[color_selgros_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.269.3780484831.ts)  
[color_smbusiness_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.269.3780484831.swift)  
[color_smbusiness_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.269.3780484831.kt)  
[color_smbusiness_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.269.3780484831.ts)  
[PlasmaTokensColor--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.269.3780484831.swift)  
[typo_mage_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.269.3780484831.swift)  
[typo_mage_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.269.3780484831.kt)  
[typo_mage_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.269.3780484831.ts)  
[typo_plasma_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.269.3780484831.swift)  
[typo_plasma_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.269.3780484831.kt)  
[typo_plasma_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.269.3780484831.ts)  
[typo_ruler_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.269.3780484831.swift)  
[typo_ruler_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.269.3780484831.kt)  
[typo_ruler_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.269.3780484831.ts)  
[typo_sage_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.269.3780484831.swift)  
[typo_sage_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.269.3780484831.kt)  
[typo_sage_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.269.3780484831.ts)  
[typo_sbermarket_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.269.3780484831.swift)  
[typo_sbermarket_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.269.3780484831.kt)  
[typo_sbermarket_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.269.3780484831.ts)  
[typo_soulmate_ios-swift--canary.269.3780484831.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.269.3780484831.swift)  
[typo_soulmate_kotlin--canary.269.3780484831.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.269.3780484831.kt)  
[typo_soulmate_react-native--canary.269.3780484831.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.269.3780484831.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.124.0-canary.269.3780484831.0
  npm install @salutejs/plasma-web@1.151.0-canary.269.3780484831.0
  # or 
  yarn add @salutejs/plasma-b2c@1.124.0-canary.269.3780484831.0
  yarn add @salutejs/plasma-web@1.151.0-canary.269.3780484831.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-269/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-269/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-269/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-269/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-269/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-269/temple-storybook/

File: packages/plasma-web/src/components/Dropdown/DropdownItem.tsx [hunk 1]
```diff
@@ -113,7 +113,7 @@ const StyledContent = styled.div`
     margin-right: 0.375rem;
 `;
 const StyledText = styled.span`
-    ${applyEllipsis}
+    ${applyEllipsis};
```
Comment (Yeti-or) [inline packages/plasma-web/src/components/Dropdown/DropdownItem.tsx:116]: а ты уверен что здесь нужна точка с запятой?

как выглядит итоговый css можешь найти ?

File: packages/plasma-web/src/components/Dropdown/keyboardNavigation/utils.ts [hunk 2]
```diff
@@ -125,17 +125,6 @@ export const getUpdatedIndex = (action: Actions, index: number, items: DropdownN
             newIndex = index;
     }
 
-    let item = items[newIndex];
-
-    while (item.isDisabled) {
```
Comment (Yeti-or) [inline packages/plasma-web/src/components/Dropdown/keyboardNavigation/utils.ts:130]: ммм то есть теперь стрлочками можно на disabled встать?

Comment (Yakutoc) [reply packages/plasma-web/src/components/Dropdown/keyboardNavigation/utils.ts:130]: Да, reader же навигацию осуществляет через них. 

Я про это вчера и говорил, и показывал на звонке. )

File: packages/plasma-web/src/components/Dropdown/DropdownItem.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-web/src/components/Dropdown/DropdownItem.tsx:116]: Здесь ругался stylelint я и поправил.

File: packages/plasma-web/src/components/Dropdown/keyboardNavigation/utils.ts (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-web/src/components/Dropdown/keyboardNavigation/utils.ts:130]: ну мы говорили с точки зрения ридера, а без этого кода ридер проскакивает disabled?

потому что с точки зрения клавиатурной навигации без ридера это выглядит страно, как будто я нажимаю стрелочку и никуда не попадаю

Comment (Yakutoc) [reply packages/plasma-web/src/components/Dropdown/keyboardNavigation/utils.ts:130]: >  а без этого кода ридер проскакивает disabled?

Да, ридер переходит к следующему элементу.

Comment (Yakutoc) [reply packages/plasma-web/src/components/Dropdown/keyboardNavigation/utils.ts:130]: > потому что с точки зрения клавиатурной навигации без ридера это выглядит страно, как будто я нажимаю стрелочку и никуда не попадаю

У нас 2 варианта развития события: 

Мы убираем стили отвечающие за disabled 

```js
 ${({ $disabled }) =>
        $disabled &&
        css`
            &,
            &:hover,
            &:focus,
            &:active {
                background-color: transparent;
            }
        `}
```   

и тогда у нас визуально видна навигация по списку и всем элементам. 

Reader попадая на выключенный элемент сообщит пользователю, что этот элемент не доступен.    

---------------------

Убираем весь код, что я добавил и используем aria-hidden={isDisabled}. 

Reader не сможет попасть на выключенный элемент, однако счет кол-ва элементов в списке будет верным.

Comment (Yakutoc) [reply packages/plasma-web/src/components/Dropdown/keyboardNavigation/utils.ts:130]: @Yeti-or @neretin-trike 

Пообщался со знающими людьми ) сказали первый вариант самое оно.

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-269/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-269/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-269/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-269/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-269/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-269/temple-storybook/

**Review by neretin-trike — APPROVED** [review]

**Review by Yeti-or — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-269/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-269/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-269/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-269/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-269/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-269/temple-storybook/

**Pull request merged**